crates/driver/src/domain/competition/settled/mod.rs

```diff
@@ -0,0 +1,275 @@
+use {
+    super::{
+        auction,
+        order::{self, Side},
+    },
+    crate::{domain::eth, util::conv::u256::U256Ext},
+    number::conversions::big_rational_to_u256,
+};
+
+/// Settlement in an onchain settleable form and semantics, aligned with what
+/// the settlement contract expects.
+#[derive(Debug, Clone)]
+pub struct Settlement {
+    trades: Vec<Trade>,
+}
+
+impl Settlement {
+    pub fn new(trades: Vec<Trade>) -> Self {
+        Self { trades }
+    }
+
+    /// Score of a settlement as per CIP38
+    ///
+    /// Score of a settlement is a sum of scores of all user trades in the
+    /// settlement.
+    ///
+    /// Settlement score is valid only if all trade scores are valid.
+    ///
+    /// Denominated in NATIVE token
+    pub fn score(&self, prices: &auction::NormalizedPrices) -> Result<eth::TokenAmount, Error> {
+        self.trades
+            .iter()
+            .map(|trade| trade.score(prices))
+            .try_fold(eth::TokenAmount(eth::U256::zero()), |acc, score| {
+                score.map(|score| acc + score)
+            })
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct Trade {
```

Comment:
nit: if moved to `competition::solution` visibility can be limited to `pub(super)` (to avoid having too many `Trade` structs in scope)

Reply:
resolved.